### ISSUE TITLE
Unit test Karma/jasmine: error while loading json language files

### ISSUE THREAD

**Issue by aelmanaa** [ocombe/ng2-translate#323]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => check the FAQ and search github for a similar issue or PR before submitting
[ ] support request => check the FAQ and search github for a similar issue before submitting
[ ] feature request
```

**Current behavior**
while running the unit tests with angular-cli ( command ng-test) , my tests fail because the loader of Karma doesn t handle correcterly the /i18n/*.json files. The following exception is returned:

Uncaught SyntaxError: Unexpected token :  at src/assets/i18n/en.json

**Expected/desired behavior**
This behavior occurs only when I try to run unit tests. Otherwise my application works perfectly fine.  ng2-translate should be compatible with karma/jasmine and we shouldn t get exception while loading *json files

**Reproduction of the problem**
If the current behavior is a bug or you can illustrate your feature request better with an example, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:btpW3l0jr5beJVjohy1Q).

run a karma unit test on a component containing the translate filter

**What is the expected behavior?**

**What is the motivation / use case for changing the behavior?**
I cannot test anymore my components which use ng translate

**Please tell us about your environment:**

* **ng2-translate version:** 3.1.3

* **Angular version:** 2.1.0
 
* **Browser:** [Chrome] 
 
* **Language:** [TypeScript 2.0]

**Comment by ocombe** [comment]
Hello,

karma serves files from a different route than your application does.

You have two options to resolve this problem:
- use MockBackend to simulate the webserver for the http request that will load the i18n json file, and return it from its real location (using node require)
- define the correct the assets location (the one that karma will serve) when you define the TranslateLoader in the TranslateModule

**Comment by aelmanaa** [comment]
Hello ocombe,

thank you very much for your answer.  Indeed I m not planning to test the loader but I thought that ng2-translate will be supported by the loader.
You ve just suggested a nice workaround , I think it should be documented since I m pretty sure that everyone who uses ng2-translate and tries to unit test a component has encountered the same issue

**Comment by ocombe** [comment]
That's true. I should add a test with that and it could serve as a documentation.

**Comment by aelmanaa** [comment]
Thanks a lot for this library and for your help as well

**Comment by CalebVuorinen** [comment]
+1

**Comment by mmarimuthu** [comment]
Hello, 
can someone provide an example translate application with unit test (i.e., test the translated values from json).

**Comment by xmeng1** [comment]
+1

**Comment by xmeng1** [comment]
https://github.com/ngx-translate/core/issues/471

**Comment by shravansofts** [comment]
When I'm trying to fetch a DOM element string, i'm getting whitespace string like **'                   '**
Im using translate pipe for translate
